accord-core/src/main/java/accord/impl/SimpleProgressLog.java

```diff
@@ -668,242 +710,123 @@ State recordExecute(TxnId txnId)
         }
 
         @Override
-        public void executedOnAllShards(Command command, Set<Id> persistedOn)
-        {
-            State state = ensure(command.txnId());
-            state.home().executedOnAllShards(node, command, persistedOn);
-        }
-
-        @Override
-        public void waiting(PartialCommand blockedByCommand, Keys someKeys)
-        {
-            TxnId blockedBy = blockedByCommand.txnId();
-            if (!blockedByCommand.hasBeen(Executed))
-                ensure(blockedBy).recordBlocking(blockedByCommand, someKeys);
-        }
-    }
-
-    @Override
-    public void run()
-    {
-        for (Instance instance : instances)
-        {
-            // TODO: we want to be able to poll others about pending dependencies to check forward progress,
-            //       as we don't know all dependencies locally (or perhaps any, at execution time) so we may
-            //       begin expecting forward progress too early
-            // state map may be updated during iteration, so need to clone the values set
-            new ArrayList<>(instance.stateMap.values()).forEach(state -> state.update(node));
-        }
-    }
-
-    static class CoordinateApplyAndCheck extends AsyncFuture<Void> implements Callback<ApplyAndCheckOk>
-    {
-        final TxnId txnId;
-        final HomeState state;
-        final Set<Id> waitingOnResponses;
-
-        static Future<Void> applyAndCheck(Node node, TxnId txnId, Command command, HomeState state)
-        {
-            CoordinateApplyAndCheck coordinate = new CoordinateApplyAndCheck(txnId, state);
-            Topologies topologies = node.topology().preciseEpochs(command.txn(), command.executeAt().epoch);
-            state.globalNotPersisted.retainAll(topologies.nodes()); // we might have had some nodes from older shards that are now redundant
-            node.send(state.globalNotPersisted, id -> new ApplyAndCheck(id, topologies,
-                                                                        command.txnId(), command.txn(), command.homeKey(),
-                                                                        command.savedDeps(), command.executeAt(),
-                                                                        command.writes(), command.result(),
-                                                                        state.globalNotPersisted),
-                      coordinate);
-            return coordinate;
-        }
-
-        CoordinateApplyAndCheck(TxnId txnId, HomeState state)
+        public void durableLocal(TxnId txnId)
         {
-            this.txnId = txnId;
-            this.state = state;
-            this.waitingOnResponses = new HashSet<>(state.globalNotPersisted);
+            State state = ensure(txnId);
+            state.global().durableLocal(node);
         }
 
         @Override
-        public void onSuccess(Id from, ApplyAndCheckOk response)
+        public void durable(Command command, @Nullable Set<Id> persistedOn)
```

Comment:
This is really globally durable?

Reply:
Maybe there's some improvements to make again to terminology, but durable is taken to mean reached our global durability requirements. Similarly, `Durability.Durable` refers to this global durability, but offers a lower `Local` level of _durability_.